test/integration/integration.test.ts

```diff
@@ -0,0 +1,533 @@
+  let c9: MockASE
+  let hlb: MockASE
+
+  beforeAll(async () => {
+    c9 = await MockASE.create(C9_CONFIG)
+    hlb = await MockASE.create(HLB_CONFIG)
+  })
+
+  afterAll(async () => {
+    c9.shutdown()
+    hlb.shutdown()
+  })
+
+  describe('Open Payments Flow', (): void => {
+    const receiverWalletAddressUrl =
+      'http://host.docker.internal:4000/accounts/pfry'
+    const senderWalletAddressUrl =
+      'http://host.docker.internal:3000/accounts/gfranklin'
+
+    let receiverWalletAddress: WalletAddress
+    let senderWalletAddress: WalletAddress
+    let accessToken: string
+    let incomingPayment: IncomingPayment
+    let quote: Quote
+    let outgoingPaymentGrant: PendingGrant
+    let grantContinue: Grant
+    let outgoingPayment: OutgoingPayment
+
+    test('Can Get Existing Wallet Address', async (): Promise<void> => {
+      receiverWalletAddress = await c9.opClient.walletAddress.get({
+        url: receiverWalletAddressUrl
+      })
+      senderWalletAddress = await c9.opClient.walletAddress.get({
+        url: senderWalletAddressUrl
+      })
+
+      expect(receiverWalletAddress.id).toBe(
+        receiverWalletAddressUrl.replace('http', 'https')
+      )
+      expect(senderWalletAddress.id).toBe(
+        senderWalletAddressUrl.replace('http', 'https')
+      )
+    })
+
+    test('Can Get Non-Existing Wallet Address', async (): Promise<void> => {
+      const notFoundWalletAddress =
+        'https://host.docker.internal:4000/accounts/asmith'
+
+      const handleWebhookEventSpy = jest.spyOn(
+        hlb.integrationServer.webhookEventHandler,
+        'handleWebhookEvent'
+      )
+
+      // Poll in case the webhook response to create wallet address is slow,
+      // but initial request may very well resolve immediately.
+      const walletAddress = await poll(
+        async () =>
+          c9.opClient.walletAddress.get({
+            url: notFoundWalletAddress
+          }),
+        (responseData) => responseData.id === notFoundWalletAddress,
+        5,
+        0.5
+      )
+
+      assert(walletAddress)
+      expect(walletAddress.id).toBe(notFoundWalletAddress)
+      expect(handleWebhookEventSpy).toHaveBeenCalledWith(
+        expect.objectContaining({
+          type: WebhookEventType.WalletAddressNotFound,
+          data: expect.objectContaining({
+            walletAddressUrl: notFoundWalletAddress
+          })
+        })
+      )
+    })
+
+    test('Grant Request Incoming Payment', async (): Promise<void> => {
+      const grant = await c9.opClient.grant.request(
+        {
+          url: receiverWalletAddress.authServer
+        },
+        {
+          access_token: {
+            access: [
+              {
+                type: 'incoming-payment',
+                actions: ['create', 'read', 'list', 'complete']
+              }
+            ]
+          }
+        }
+      )
+
+      assert(!isPendingGrant(grant))
+      accessToken = grant.access_token.value
```

Comment:
Note that these tests dont have too much in the way of assertions. I am open to feedback on more meaningful assertions if you see an opportunity, but the flow completing is really the whole test. 

Most of the assertions are based around making the code typesafe, providing some minimal level of validation that the response isn't bad (which the subsequent test would catch anyways), or verifying we received webhooks. Many of these really just make it easier to see where things went wrong rather than ensuring everything works - whether or not it works really boils down to if they all pass and they wont if each step doesn't complete without error. Our service unit tests are really the place to make more detailed assertions about what we expect to see. We can validate some additional things here like receiving 

In this way its actually kind of misleading to have separate tests per request but that's something I elaborate on and will tackle in another issue (https://github.com/interledger/rafiki/issues/2529).